index.bs

```diff
@@ -101,21 +118,190 @@ spec: webidl
 </section>
 <section>
   <h2 id="permission-descriptor">
-    Permission descriptor
+    Descriptions of permission requests
   </h2>
   <pre class='idl' title=''>
     dictionary PermissionDescriptor {
       required PermissionName name;
     };
   </pre>
   <p>
-    A permission is described by a name and other properties that depend on the
-    name. The simplest permissions require only a name, but some others have
-    more detailed structure that requires more information to describe it. In
-    that case, they should define a customized <a>permission descriptor
-    type</a> dictionary that inherits from {{PermissionDescriptor}}.
+    Each powerful feature has one or more aspects that websites may request
+    permission to access. To describe these aspects, each feature defines a
+    subtype of {{PermissionDescriptor}} to be its <a>permission descriptor
+    type</a>.
   </p>
+
+  <div class="example" id="example-descriptors">
+    <p>
+      The {{"midi"}} feature has two aspects: access to normal messages, and
+      access to system exclusive messages. Thus, its permission descriptor type
+      is:
+    </p>
+    <pre highlight="idl">
+      dictionary MidiPermissionDescriptor : PermissionDescriptor {
+        boolean sysex = false;
+      };
+    </pre>
+    <p>
+      The {{"camera"}} feature lets sites access whatever cameras are connected
+      to the user's device. Thus, its descriptor type is:
+    </p>
+    <pre highlight="idl">
+      dictionary DevicePermissionDescriptor : PermissionDescriptor {
+        DOMString deviceId;
+      };
+    </pre>
+    <p>
+      General access to cameras is represented by `{name: 'camera'}`, while
+      access to a particular camera is represented by `{name: 'camera',
+      deviceId: cameraId}`.
+    </p>
+  </div>
+</section>
+<section>
+  <h2 id="permission-operations">Permission states</h2>
+  <p>
+    The user agent is responsible for tracking what powerful features each
+    <a>realm</a> has the user's permission to use. Other specifications can use
```

Comment:
I might be mistaken, but I was under the impression we were going to avoid "realms" in web specs and leave it to ES. I wonder if there is something we could use from HTML regarding security contexts?